### PR TITLE
Shows sharing links on selecting text

### PR DESCRIPTION
**Pull request by Gauravjeetsingh**
* [x]  Passed [sanity tests](http://bit.ly/sttm-sanity-tests).
* [x]  Ran `npm test` & fixed newly introduced lint errors.
* [x]  Checked console for errors.

I read the discussion in issue #122 and came with this solution.

Instead of showing the share actions on hover, it will now show when we select the text.
Even If we select a little part from the baani, it will select the entire line (including translation and transliteration) and shows it's share action links.

5notp-3uhwu

It also fixes the color change effect when we hover on the share action icon.

<!-- New to markdown? Simply put an 'x' between [ ] to check it. Like [x] this. (No spaces).

### CODE REVIEW

File: src/js/components/Baani.js [hunk 1]
```diff
@@ -114,8 +114,32 @@ export default class Baani extends React.PureComponent {
...
+      const shareDiv = baaniDiv.querySelector('.share');
+      shareDiv.style.display = 'block';
+    }
+  };
+
   componentDidMount() {
     this._scrollToHiglight();
+    const lineDivs = document.querySelectorAll('.line');
+    const ms = this.makeSelection;
+    lineDivs.forEach(function(ld) {
```
Comment (bogas04) [inline src/js/components/Baani.js]: Can we instead do it in the React side? Like event listeners on the line components

Comment (Gauravjeetsingh) [reply src/js/components/Baani.js]: Fixed @bogas04

File: src/js/components/Baani.js [hunk 2]
```diff
@@ -153,6 +173,8 @@ export default class Baani extends React.PureComponent {
             key={shabad.id}
             id={`line-${shabad.id}`}
             className="line"
+            onMouseUp={this.makeSelection}
+            onMouseDown={this.removeSelection}
```
Comment (bogas04) [inline src/js/components/Baani.js]: Wouldn't work for split view then right?

File: src/js/components/Baani.js [hunk 3]
```diff
@@ -193,8 +215,6 @@ export default class Baani extends React.PureComponent {
               onFacebookClick={this.onFacebookClick(shabad)}
               onCopyClick={this.onCopyClick(shabad)}
             />
-
-            <br />
```
Comment (bogas04) [inline src/js/components/Baani.js]: I think this was added to keep new lines in the copy pasted text.

File: src/js/components/Baani.js [hunk 4]
```diff
@@ -116,6 +116,26 @@ export default class Baani extends React.PureComponent {
     }
   };
 
+  makeSelection = e => {
+    const shareDivs = document.querySelectorAll('.share');
+    shareDivs.forEach(function(sd) {
+      sd.style.display = 'none';
+    });
```
Comment (bogas04) [inline src/js/components/Baani.js]: Use arrow function

File: src/js/components/Baani.js [hunk 5]
```diff
@@ -116,6 +116,26 @@ export default class Baani extends React.PureComponent {
...
+    if (window.getSelection().toString()) {
+      const baaniDiv = e.currentTarget;
+      var range = document.createRange();
+      range.selectNode(baaniDiv);
+      window.getSelection().removeAllRanges();
+      window.getSelection().addRange(range);
+      const shareDiv = baaniDiv.querySelector('.share');
+      shareDiv.style.display = 'block';
+    }
```
Comment (bogas04) [inline src/js/components/Baani.js]: Let's make a helper function that takes a DOM node and selects it. That way the functionality and logic isn't coupled. Makes sense?

File: src/js/components/Baani.js (hunk 3 above)
Comment (Gauravjeetsingh) [reply src/js/components/Baani.js]: I checked copy paste with and without br tag, it's same as following.

```
ਤੁਧ ਬਿਨੁ ਰੋਗੁ ਰਜਾਈਆਂ ਦਾ ਓਢਣੁ ਨਾਗ ਨਿਵਾਸਾਂ ਦੇ ਰਹਣਾ ॥
tudh bin rog rajaieeaa(n) dhaa oddan naag nivaasaa(n) dhe rahanaa ||
Without Thee, the taking over of quilt is like disease and living in the house is like living with serpents
```

File: src/js/components/Baani.js (hunk 2 above)
Comment (Gauravjeetsingh) [reply src/js/components/Baani.js]: Yeah, it doesn't work in split view. Do we want it to work in split view?

File: src/js/components/Baani.js (hunk 5 above)
Comment (Gauravjeetsingh) [reply src/js/components/Baani.js]: Fixed.

File: src/js/components/Baani.js (hunk 4 above)
Comment (Gauravjeetsingh) [reply src/js/components/Baani.js]: Removed the function. Using a separate class to show.

File: src/js/components/Baani.js (hunk 5 above)
Comment (bogas04) [reply src/js/components/Baani.js]: Let's keep this in helpers file so that other components can use as well. What do you think?

Comment (Gauravjeetsingh) [reply src/js/components/Baani.js]: Yeah, good idea. Done. :raised_hands:

**Pull request merged**